controllers/machinedeletionremediation_controller.go

```diff
@@ -70,31 +70,32 @@ type MachineDeletionRemediationReconciler struct {
 func (r *MachineDeletionRemediationReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
 	log := r.Log.WithValues("machinedeletionremediation", req.NamespacedName)
 
-	log.Info("reconciling...")
-
-	//fetch the remediation
+	var err error
 	var remediation *v1alpha1.MachineDeletionRemediation
-	if remediation = r.getRemediation(ctx, req); remediation == nil {
-		return ctrl.Result{}, nil
+	if remediation, err = r.getRemediation(ctx, req); remediation == nil || err != nil {
+		return ctrl.Result{}, err
 	}
 
-	var machine *unstructured.Unstructured
-	//Health check was done by NHC
-	if node, err := r.getNodeFromMdr(remediation); err == nil {
-		if machine, err = r.buildMachineFromNode(node); err != nil {
-			r.Log.Error(err, "failed to fetch machine of node", "node name", node.Name)
-			return ctrl.Result{}, err
-		}
-		if !hasControllerOwner(machine) {
-			r.Log.Info("ignoring remediation of machine associated to node, since the machine has no controller owner", "node name", remediation.Name)
-			return ctrl.Result{}, nil
-		}
-
-	} else { //Failed fetching the node
+	var node *v1.Node
+	if node, err = r.getNodeFromMdr(remediation); err != nil {
 		r.Log.Error(err, "failed to fetch node", "node name", remediation.Name)
 		return ctrl.Result{}, err
 	}
+	log.Info("reconciling...")
+
+	var machine *unstructured.Unstructured
+	if machine, err = r.buildMachineFromNode(node); err != nil {
+		r.Log.Error(err, "failed to fetch machine of node", "node name", node.Name)
+		return ctrl.Result{}, err
+	}
+	log.Info("node-associated machine found", "machine", machine.GetName())
+
+	if !hasControllerOwner(machine) {
+		log.Info("ignoring remediation of machine associated to node, since the machine has no controller owner", "node name", remediation.Name)
+		return ctrl.Result{}, nil
+	}
 
+	log.Info("request machine deletion", "machine", machine.GetName())
```

Comment:
Nit: redundant IMO (since we log failure at line:94, I think the log at 91 is enough)